src/main/java/org/openrewrite/gradle/RewriteRecipeAuthorAttributionTask.java

```diff
@@ -129,67 +145,151 @@ static class Attribution {
+            File recipeFile = change.getFile();
+            String recipeFqn = recipeFileNameToFqn.get(recipeFile.getName());
+            if (recipeFqn == null) {
+                continue;
+            }
+            Path targetPath = outputDir.resolve(recipeFqn + ".yml");
+            if (change.getChangeType() == ChangeType.REMOVED) {
+                Files.delete(targetPath);
+                continue;
+            }
+
+            // git commands only accept forward slashes in paths, regardless of operating system
+            String relativePath = getProject().getRootDir().toPath()
+                    .relativize(recipeFile.toPath()).toString()
+                    .replace('\\', '/');
+            BlameResult blame = g.blame().setFilePath(relativePath).call();
+            if (blame == null || blame.getResultContents() == null) {
+                continue;
+            }
+
+            List<Contributor> contributors = extractContributorsForRange(blame, 0, blame.getResultContents().size());
+            Attribution attribution = new Attribution("specs.openrewrite.org/v1beta/attribution", recipeFqn, contributors);
+            String yaml = mapper.writeValueAsString(attribution);
+
+            Files.createDirectories(targetPath.getParent());
+            Files.writeString(targetPath, yaml);
+        }
+    }
+
+    private void processYamlRecipes(Git g, InputChanges inputChanges) throws Exception {
+        YAMLMapper mapper = new YAMLMapper();
+        Path outputDir = getOutputDirectory();
 
+        for (FileChange change : inputChanges.getFileChanges(sources)) {
+            File yamlFile = change.getFile();
+            if (!yamlFile.getName().endsWith(".yml") && !yamlFile.getName().endsWith(".yaml")) {
+                continue;
+            }
+
+            if (change.getChangeType() == ChangeType.REMOVED) {
+                Path targetPath = outputDir.resolve(yamlFile.getName().replaceFirst("\\.[^.]+$", "") + ".yml");
+                Files.delete(targetPath);
+                continue;
+            }
+
+            try {
                 // git commands only accept forward slashes in paths, regardless of operating system
                 String relativePath = getProject().getRootDir().toPath()
-                        .relativize(recipeFile.toPath()).toString()
+                        .relativize(yamlFile.toPath()).toString()
                         .replace('\\', '/');
-                List<Contributor> contributors = listContributors(g, relativePath);
+                BlameResult blame = g.blame().setFilePath(relativePath).call();
 
-                Attribution attribution = new Attribution("specs.openrewrite.org/v1beta/attribution", recipeFqn, contributors);
-                String yaml = mapper.writeValueAsString(attribution);
+                List<RecipeLineNumbers> recipeLineNumbers = recipeLineNumbers(yamlFile);
+                for (RecipeLineNumbers recipe : recipeLineNumbers) {
```

Comment:
Whereas for Yaml files we need to look at the start/stop of every document, to determine the right recipe & authors.